pkg/sshutil/sshutil.go

```diff
@@ -18,34 +19,74 @@ type PubKey struct {
 	Content  string
 }
 
-// DefaultPubKeys finds ssh public keys from ~/.ssh
+func readPublicKey(f string) (PubKey, error) {
+	entry := PubKey{
+		Filename: f,
+	}
+	content, err := os.ReadFile(f)
+	if err == nil {
+		entry.Content = strings.TrimSpace(string(content))
+	} else {
+		logrus.WithError(err).Warningf("failed to read ssh public key %q", f)
+	}
+	return entry, err
+}
+
+// DefaultPubKeys returns the public key from $LIMA_HOME/_config/user.pub.
+// If that doesn't exist, it finds all ssh public keys from ~/.ssh.
+// If there aren't any, it generates a key at $LIMA_HOME/_config/user
+// and returns the corresponding public key.
 func DefaultPubKeys() []PubKey {
-	homeDir, err := os.UserHomeDir()
+	// Read $LIMA_HOME/_config/user.pub
+	configDir, err := store.LimaConfigDir()
+	privateKeyPath := filepath.Join(configDir, "user")
 	if err != nil {
 		logrus.Warn(err)
-		return nil
+	} else {
+		// Check that private key exists and then try to read public key
+		if _, err := os.Stat(privateKeyPath); err == nil {
```

Comment:
Return err if !error.Is(err, os.ErrNorExist)

Reply:
`DefaultPubKeys` doesn't return any errors; it just ignores key files it cannot read. So not sure what to do here.

In the end the user would be told to run `ssh-keygen` when `DefaultPubKeys` returns an empty list, so not sure that aborting on the first error is really helpful.